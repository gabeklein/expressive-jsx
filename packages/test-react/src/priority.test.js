const test = require("./_adapter");

test("competing style priority", `
  Bar: { color: red };

  () => do {
    Bar: { color: blue }
    Foo: {
      Bar: { color: orange }
    }
    
    <Foo>
      <Bar />
    </Foo>
  };
`);

test("conditional priority should be higher", `
  () => do {
    if(checked)
      test: {
        color: red;
      }

    test: {
      color: green;
    }

    <test />
}
`);

test("prioritize external (capital-letter) define", `
  const Foo = () => do {
    forward: className;
    color: red;

    <this>Hello</this>
  }

  const Bar = () => do {
    Foo: {
      color: blue;
    }
    
    <Foo />
  }
`);