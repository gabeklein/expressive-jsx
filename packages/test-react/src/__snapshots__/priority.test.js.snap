// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`competing style priority 1`] = `
";

() => create(Foo, {}, create(Bar, {
  className: \\"Bar_i3g\\"
}));

CSS.put(\`
	/* 2 */
	.Bar_i3g { color: red }
\`);"
`;

exports[`competing style priority 2`] = `
";

() => <Foo><Bar className=\\"Bar_i3g\\" /></Foo>;

CSS.put(\`
	/* 2 */
	.Bar_i3g { color: red }
\`);"
`;

exports[`conditional priority should be higher 1`] = `
"() => create(\\"div\\", {
  className: use(\\"element_26b\\", !!checked && \\"if_checked_6ac\\")
}, create(\\"div\\", {
  className: \\"test_2ng test_6jg\\"
}));

CSS.put(\`
	/* 2 */
	.test_6jg { color: green }
	/* 4 */
	.element_26b.if_checked_6ac .test_2ng { color: red }
\`);"
`;

exports[`conditional priority should be higher 2`] = `
"() => <div className={use(\\"element_26b\\", !!checked && \\"if_checked_6ac\\")}><div className=\\"test_2ng test_6jg\\" /></div>;

CSS.put(\`
	/* 2 */
	.test_6jg { color: green }
	/* 4 */
	.element_26b.if_checked_6ac .test_2ng { color: red }
\`);"
`;

exports[`prioritize external (capital-letter) define 1`] = `
"const Foo = ({
  className
}) => create(\\"div\\", {
  className: use(\\"Foo_8ef\\", className)
}, \\"Hello\\");

const Bar = () => create(Foo, {
  className: \\"Foo_124\\"
});

CSS.put(\`
	/* 1 */
	.Foo_8ef { color: red }
	/* 2 */
	.Foo_124 { color: blue }
\`);"
`;

exports[`prioritize external (capital-letter) define 2`] = `
"const Foo = ({
  className
}) => <div className={use(\\"Foo_8ef\\", className)}>Hello</div>;

const Bar = () => <Foo className=\\"Foo_124\\" />;

CSS.put(\`
	/* 1 */
	.Foo_8ef { color: red }
	/* 2 */
	.Foo_124 { color: blue }
\`);"
`;
